opengrok-web/src/main/java/org/opengrok/web/CorsFilter.java

```diff
@@ -0,0 +1,69 @@
+package org.opengrok.web;
+
+import javax.ws.rs.container.*;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.Provider;
+import java.io.IOException;
+
+@Provider
+@PreMatching
+public class CorsFilter implements ContainerRequestFilter, ContainerResponseFilter {
+
+    /**
+     * Method for ContainerRequestFilter.
+     */
+    @Override
+    public void filter(ContainerRequestContext request) throws IOException {
+
+        // If it's a preflight request, we abort the request with
+        // a 200 status, and the CORS headers are added in the
+        // response filter method below.
+        if (isPreflightRequest(request)) {
+            request.abortWith(Response.ok().build());
+            return;
+        }
+    }
+
+    /**
+     * A preflight request is an OPTIONS request
+     * with an Origin header.
+     */
+    private static boolean isPreflightRequest(ContainerRequestContext request) {
+        return request.getHeaderString("Origin") != null
+                && request.getMethod().equalsIgnoreCase("OPTIONS");
```

Comment:
prefight request per spec is not only OPTIONS with origin but also with special headers

> It is an OPTIONS request, using three HTTP request headers: Access-Control-Request-Method, Access-Control-Request-Headers, and the Origin header. (https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request)

Reply:
And in our special case we only support GET for the selected paths so essentially the prefight request would not be needed at all.

Reply:
I don't see pre-flight requests being handled by the current implementation.

Reply:
Cause it's not needed.

Reply:
how so ?

Reply:
Based on the doc I sent in other responses here https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#Simple_requests. The cors has two request categories

1. simple requests
2. prefighted requests

Quoting the mozilla spec: **simple request meets all the following conditions**:


    1. The only allowed methods are:
        GET
        HEAD
        POST
    2 .Apart from the headers set automatically by the user agent (for example, Connection, User-Agent, or any of the other headers with names defined in the Fetch spec as a “forbidden header name”), the only headers which are allowed to be manually set are those which the Fetch spec defines as being a “CORS-safelisted request-header”, which are:
        Accept
        Accept-Language
        Content-Language
        Content-Type (but note the additional requirements below)
        DPR
        Downlink
        Save-Data
        Viewport-Width
        Width
    3. The only allowed values for the Content-Type header are:
        application/x-www-form-urlencoded
        multipart/form-data
        text/plain
    4. No event listeners are registered on any XMLHttpRequestUpload object used in the request; these are accessed using the XMLHttpRequest.upload property.
    5. No ReadableStream object is used in the request.


For all paths which should be covered with CORS on our side, all uses GET method (1), don't have body (so Content-Type header is needless) (2+3), don't upload files (4), don't need to use streaming api for receiving simple html page (5). Therefore prefight request aren't necessary.